src/05-derivatives/01-introduction.md

```diff
@@ -0,0 +1,282 @@
+    under which it stores all of its outputs. Different components of a pipeline can, 
+    however, also be stored under different subfolders. There are few restrictions on
+    the directory names; it is RECOMMENDED to use the format `<pipeline>-<variant>` in 
+    cases where it is anticipated that the same pipeline will output more than one variant (e.g.,
+    `AFNI-blurring`, `AFNI-noblurring`, etc.). For the sake of consistency, the
+    subfolder name SHOULD be the `PipelineDescription.Name` field in
+    `data_description.json`, optionally followed by a hyphen and a suffix (see
+    below).
+
+    Example of derivatives with one directory per pipeline:
+
+    ```Plain
+    <dataset>/derivatives/fmriprep-v1.4.1/sub-0001
+    <dataset>/derivatives/spm/sub-0001
+    <dataset>/derivatives/vbm/sub-0001
+    ```
+
+    Example of a pipeline with split derivative directories:
+
+    ```Plain
+    <dataset>/derivatives/spm_preproc/sub-0001
+    <dataset>/derivatives/spm_stats/sub-0001
+    ```
+
+    Example of a pipeline with nested derivative directories:
+
+    ```Plain
+    <dataset>/derivatives/spm_preproc/sub-0001
+    <dataset>/derivatives/spm_preproc/derivatives/spm_stats/sub-0001
+    ```    
+
+
+1.  As a standalone dataset independent of the source (raw or derived) BIDS
+    dataset. This way of specifying derivatives is particularly useful when the
+    source dataset is provided with read-only access, and for publishing
+    derivatives as independent bodies of work, or for describing derivatives
+    that were created from more than one source dataset. The `sourcedata/`
+    subdirectory MAY be used to include the source dataset(s) that were used
+    to generate the derivatives.
+    Likewise, any code used to generate the derivatives from the source data
+    MAY be included in the `code/` subdirectory.
+
+The rest of the Derivatives specification assumes Case 1, but Case 2 applies
+after removing `/derivatives/<pipeline_name>` from template names. In both cases
+every derivatives dataset is considered a BIDS dataset and must include a
+`dataset_description.json` file at the root level. Consequently, files should be
+organized to comply with BIDS to the full extent possible (i.e., unless
+explicitly contradicted below). Any subject-specific derivatives should be
+housed within each subject’s directory; if session-specific derivatives are
+generated, they should be deposited under a session subdirectory within the
+corresponding subject directory; and so on.
+
+## Derived dataset and pipeline description
+
+As for any BIDS dataset a `dataset_description.json` file MUST be found at the
+top level of the particular pipeline:
+`<dataset>/derivatives/<pipeline_name>/dataset_description.json`
+
+In addition to raw BIDS datasets, derived BIDS datasets include the following
+required or recommended `dataset_description.json` keys (a dot in the Key name
+denotes a key in a subdictionary):
```

Comment:
We have a sentence, followed by a table, followed by an example. I'm not sure that inserting an example before the table is clearer, rather than cluttering.

For that particular issue, it might be better to improve the PyBIDS error message.

WDYT?

Reply:
Improving the pyBIDS error message is closer to the issue being experienced, so your suggestion sounds reasonable, thanks!